### DISCUSSION TITLE
❓ Questions / Help / Support

### DISCUSSION THREAD

**Discussion by kuruhuru** [snakers4/silero-vad#194] [Show and tell]
## ❓ Questions and Help

Hello!
I got strange error. Please, advice any fix

I have a model that works and successfully learns. No errors.

But after I run your code:

USE_ONNX = False # change this to True if you want to test onnx model
if USE_ONNX:
    !pip install -q onnxruntime
  
vad_model, ut = torch.hub.load(repo_or_dir='snakers4/silero-vad',
                              model='silero_vad',
                              force_reload=True,
                              onnx=USE_ONNX)

My model starts to generate errors when learning.

   110 
    111                 model.zero_grad()
--> 112                 loss.backward()
    113 
    114                 optimizer.step()

/usr/local/lib/python3.7/dist-packages/torch/_tensor.py in backward(self, gradient, retain_graph, create_graph, inputs)
    361                 create_graph=create_graph,
    362                 inputs=inputs)
--> 363         torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
    364 
    365     def register_hook(self, hook):

/usr/local/lib/python3.7/dist-packages/torch/autograd/__init__.py in backward(tensors, grad_tensors, retain_graph, create_graph, grad_variables, inputs)
    173     Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
    174         tensors, grad_tensors_, retain_graph, create_graph, inputs,
--> 175         allow_unreachable=True, accumulate_grad=True)  # Calls into the C++ engine to run the backward pass
    176 
    177 def grad(

RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

**Comment by snakers4** [comment]
Hi,

Our models are not designed for learning.
Also please note that somewhere in our code we most likely disable the gradients during a forward pass.

As for testing how our models efffect the learning of other models - this is out of scope of this repository.
You may want to look through / ask around the discussions, I remember someone using the model as a feature extractor.

**Reply by kuruhuru** [reply]
Thank you for the answer. 

torch.set_grad_enabled(True)

This works ok after installation.

**Reply by snakers4** [reply]
Then it is most likely the disabled gradients
If you are using the models are a feature extractor, make sure not to use our convenience functions, because they may also contain code like

```
with torch.no_grad()
...
```
or similar
